HPCC-16521 CMemoryUsageReporter crash from klogctl kernel bug

mckellyln opened the pull request:
Signed-off-by: Mark Kelly <mark.kelly@lexisnexis.com>

mckellyln committed:
HPCC-16521 CMemoryUsageReporter crash from klogctl bug

Signed-off-by: Mark Kelly <mark.kelly@lexisnexis.com>

mckellyln commented:
Original PR was: https://github.com/hpcc-systems/HPCC-Platform/pull/9445

HPCCSmoketest commented:
Automated Smoketest
Sha: 6ef8eeee629939aca1baec086b87be1dfb3c8950
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.529 seconds |

Install hpccsystems-platform-community_6.0.11-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unittest result:

| total | passed | failed | timeout | 
|---|---|---|---|
|87 |86 |0 |1 |

Error(s):
CcdFileTest run timeout



HPCC Stop: OK
HPCC Uninstall: OK

mckellyln commented:
Adding note from previous PR -
This update fixes a kernel bug where klogctl() can read more than was reqested/allocated by allocating 2x more than it requests. It also only reads the last 4k of the kernel ring buffer as reading the entire buffer seems excessive when we are mostly looking for changes since last time read. This update also fixes an issue with linked https://track.hpccsystems.com/browse/HPCC-15703 in that when klogctl() returned fewer bytes than previously the code would assume all data returned is new and would repeat printing kernel msgs. This update remembers the last line crc and searches for that to know what new data is present.
